src/rpc_server.rs

```diff
@@ -127,19 +136,42 @@ impl EmitterServer for EmitterRpc {
     }
 
     async fn delete(&self, search_key: RpcSearchKey) -> Result<bool, Error> {
-        if self.state.remove(&search_key).is_some() {
+        if self.state.cell_states.remove(&search_key).is_some() {
             let handle = self.cell_handles.remove(&search_key).unwrap();
             handle.1.abort();
             return Ok(true);
         }
         Ok(false)
     }
 
-    async fn info(&self) -> Result<Vec<(RpcSearchKey, ScanTip)>, Error> {
-        Ok(self
-            .state
-            .iter()
-            .map(|kv| (kv.key().clone(), kv.value().clone()))
-            .collect::<Vec<_>>())
+    async fn info(&self) -> Result<State, Error> {
+        Ok(self.state.clone())
+    }
+
+    async fn header_sync_start(&self, number: BlockNumber) -> Result<bool, Error> {
+        let current =
+            unsafe { (&*self.state.header_state.0 .0.load(Ordering::Acquire)).block_number };
+        if number < current {
+            Ok(false)
+        } else {
+            let new_header = self.client.get_header_by_number(number).await?;
+            let new_tip = {
+                IndexerTip {
+                    block_hash: new_header.hash,
+                    block_number: new_header.inner.number,
+                }
+            };
+            let raw = self
+                .state
+                .header_state
+                .0
+                 .0
```

Comment:
extra space

Reply:
It seems that the default formatting is like this and is not changed manually

Reply:
ok